src/core/session.ts

```diff
@@ -151,8 +152,10 @@ export class Session implements NavigatorDelegate {
 
   // Form submit observer delegate
 
-  willSubmitForm(form: HTMLFormElement, submitter?: HTMLElement) {
-    return true
+  willSubmitForm(form: HTMLFormElement, submitter?: HTMLElement): boolean {
+    const methodIsHttp = !!fetchMethodFromString(submitter?.getAttribute("formmethod") || form.method)
```

Comment:
I wonder if this condition should just be moved into FormSubmitObserver? I think Turbo should probably just outright ignore any form with `formmethod="dialog"`.

Reply:
I've moved the HTTP method conditional up into the FormSubmitObserver.

> I think Turbo should probably just outright ignore submit events from any form with formmethod="dialog".

I think there is utility in passing through `submit` events. Consider an example:

```html
<dialog>
  <form method="post" action="/comments">
    <textarea name="body"></textarea>

    <button type="submit">Save</button>
    <button type="submit" formmethod="dialog">Discard</button>
  </form>
</dialog>
```

Despite a button containing an element with `[formmethod]`, we'd still want to submit the `<form>` if the `<button type="submit">Save</button>` is clicked.

Reply:
Ah, yep, you're right 👍

What do you think about comparing the value against `"dialog"` instead of searching for it in the list of HTTP methods? Are there any other special methods we ought to account for?

Reply:
I did my best to explain it in the commit message, but I could elaborate more:

> In practice, `<form>` elements will only ever declare method="get" or
method="post", but given the addition of method="dialog" and the
fact that the fetchMethodFromString and the FetchMethod enum type
exist, we might as well be careful and future-proof.

Does that feel pre-mature?

Reply:
Yeah, I think it's OK to take [the form submission spec](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#form-submission-2) at its word for now:

> 23. If the value of method is dialog then jump to the submit dialog steps.
>
> Otherwise, select the appropriate row in the table below based on the value of scheme as given by the first cell of each row. Then, select the appropriate cell on that row based on the value of method as given in the first cell of each column. Then, jump to the steps named in that cell and defined below the table.
> […]

Reply:
Done!